### PR TITLE
Import rocket's GeneratorApp directly instead of copying

### PR DESCRIPTION
**Pull request by edwardcwang**
It's currently out of date and can cause casting errors

### CODE REVIEW

File: src/main/scala/Generator.scala [hunk 1]
```diff
@@ -5,38 +5,7 @@ import freechips.rocketchip.config.{Parameters, Config}
...
-  def generateAnno {
-    val annoFile = new File(names.targetDir, s"$longName.anno")
-    val afw = new FileWriter(annoFile)
-    afw.write(JsonProtocol.serialize(circuit.annotations.map(_.toFirrtl)))
-    afw.close()
-  }
-}
+import freechips.rocketchip.util.GeneratorApp
 
 object Generator extends GeneratorApp {
```
Comment (jwright6323) [inline src/main/scala/Generator.scala]: Does this need to be ```object Generator extends freechips.rocketchip.util.GeneratorApp {```?

**Comment by jwright6323** [comment]
Do we need:
https://github.com/ucb-bar/testchipip/commit/4aedbaf8c32805350f59ecd8b82bd833810fec80#diff-8abc7f24bbe4cfc68dcbdcd13daa9ce4R6

**Comment by edwardcwang** [comment]
Done.

File: src/main/scala/Generator.scala (hunk 1 above)
Comment (edwardcwang) [reply src/main/scala/Generator.scala]: I think once it's been imported we don't need to refer to the fully-qualified/long name.

**Comment by edwardcwang** [comment]
Requires a diff in project-template, will PR later with a bump to testchipip

```diff
diff --git a/src/main/scala/example/TestHarness.scala b/src/main/scala/example/TestHarness.scala
index e67a189..d97b89e 100644
--- a/src/main/scala/example/TestHarness.scala
+++ b/src/main/scala/example/TestHarness.scala
@@ -3,7 +3,7 @@ package example
 import chisel3._
 import freechips.rocketchip.diplomacy.LazyModule
 import freechips.rocketchip.config.{Field, Parameters}
-import testchipip.GeneratorApp
+import freechips.rocketchip.util.GeneratorApp
```

**Comment by edwardcwang** [comment]
@zhemao It seems to work (elaborates and builds) with the above changes to project-template. Tested as far as building fesvr (but this fix shouldn't affect building fesvr anyway, since it's a scala-land change)

**Comment by zhemao** [comment]
Is there a reason you omitted `topModuleClass` in `longName`? I would rather the makefile not have to be changed. Sometimes we might want to synthesize different top-level modules in a package.

**Comment by edwardcwang** [comment]
@zhemao They must have changed it in upstream rocket-chip - the `longname` in this patch is copied directly from upstream rocket-chip (in fact the `Generator` class will not compile without it).

**Comment by zhemao** [comment]
The makefile in project-template is different from the one in rocket-chip. That's deliberate. I know you need to define longName. I'm saying you should define it to the same as it was before.

**Comment by edwardcwang** [comment]
I see; trying a fix.

**Comment by edwardcwang** [comment]
New diff in project-template:

```diff
diff --git a/Makefrag b/Makefrag
index 4a6be9f..62f6efc 100644
--- a/Makefrag
+++ b/Makefrag
@@ -25,7 +25,7 @@ include $(testchip_dir)/Makefrag
 CHISEL_ARGS ?=
 
 FIRRTL_FILE=$(build_dir)/$(PROJECT).$(MODEL).$(CONFIG).fir
-ANNO_FILE=$(build_dir)/$(PROJECT).$(MODEL).$(CONFIG).anno
+ANNO_FILE=$(build_dir)/$(PROJECT).$(MODEL).$(CONFIG).anno.json
 VERILOG_FILE=$(build_dir)/$(PROJECT).$(MODEL).$(CONFIG).v
 
 $(FIRRTL_FILE) $(ANNO_FILE): $(SCALA_SOURCES) $(bootrom_img) $(FIRRTL_JAR)
diff --git a/src/main/scala/example/TestHarness.scala b/src/main/scala/example/TestHarness.scala
index e67a189..20b7dc0 100644
--- a/src/main/scala/example/TestHarness.scala
+++ b/src/main/scala/example/TestHarness.scala
@@ -3,7 +3,7 @@ package example
 import chisel3._
 import freechips.rocketchip.diplomacy.LazyModule
 import freechips.rocketchip.config.{Field, Parameters}
-import testchipip.GeneratorApp
+import freechips.rocketchip.util.GeneratorApp
 
 case object BuildTop extends Field[(Clock, Bool, Parameters) => ExampleTopModule[ExampleTop]]
 
@@ -21,6 +21,7 @@ class TestHarness(implicit val p: Parameters) extends Module {
 }
 
 object Generator extends GeneratorApp {
+  val longName = names.topModuleProject + "." + names.topModuleClass + "." + names.configs
   generateFirrtl
   generateAnno
 }
```

**Comment by edwardcwang** [comment]
@zhemao I added the model back like the old code and it appears to work with the corresponding change in project-template (plus an update to the annotations format).

**Pull request merged**